src/domain/LoginViewModel.js

```diff
@@ -93,4 +93,47 @@ export class LoginViewModel extends ViewModel {
             this._sessionContainer.deleteSession();
         }
     }
+
+    /**
+     * This function will track change event into the loadViewModal,
+     * and then handle it by emitChange (isBusy) in this model view
+     * @returns {Object} the load view model subscription
+     */
+    _trackLoadViewModelChange () {
+        const  _onchangeHandler = () => {
+            if (!this._loadViewModel.loading) {
+                this._loadViewModelSubscription = this.disposeTracked(this._loadViewModelSubscription);
+            }
+            this.emitChange("isBusy");
+        }
+
+        return this.track(this._loadViewModel.disposableOn("change", _onchangeHandler));
+    }
+
+    /**
+     * This function requests the login flows from _sessionContainer using SessionLoadViewModel
+     * @param {String} homeServer - the homeserver url
+     */
+    _requestLoginFlows (homeServer) {
+        this._loadViewModel  = this.track(new SessionLoadViewModel({
+            createAndStartSessionContainer: () => {
+                this._sessionContainer = this._createSessionContainer();
+                this._sessionContainer.requestSupportedLoginFlows(homeServer);
+                return this._sessionContainer;
+            },
+            ready: sessionContainer => {
+                this._sessionContainer = null;
+                this._mapLoginView (sessionContainer.supportedLoginFlows);
+            },
+            homeServer,
+        }));
+        this._loadViewModel.start();
+        this.emitChange("loadViewModel");
+        this._loadViewModelSubscription = this._trackLoadViewModelChange();
+    }
+
+    _mapLoginView(loginFlows) {
+        // TODO: map view based on the loginFlows.
+    }
+
```

Comment:
Next step : is to use this function with loaded `loginFlows`  to set new state in into `LoginViewModel` that describe the state of  `loginView` I think there is no need to make a new view for every login flow. 
what do you think ?! @bwindels